service/history/replicationTaskProcessor.go

```diff
@@ -398,7 +400,7 @@ func (p *ReplicationTaskProcessorImpl) convertTaskToDLQTask(
 				TaskId:       replicationTask.GetSourceTaskId(),
 				TaskType:     enumsspb.TASK_TYPE_REPLICATION_HISTORY,
 				FirstEventId: firstEvent.GetEventId(),
-				NextEventId:  lastEvent.GetEventId(),
+				NextEventId:  lastEvent.GetEventId() + 1,
```

Comment:
Definitely needs a comment.  Looks like `FirstEventId` is inclusive and `NextEventId` is exclusive.  Might not be obvious for new developers.